src/status_im/chat/models/mentions.cljs

```diff
@@ -213,58 +220,40 @@
+                                            (-> chat-id
+                                                contact.db/public-key->new-contact
+                                                contact.db/enrich-contact)))
+
+      (= chat-type constants/community-chat-type)
+      (merge mentionable-users
+             (mentionable-contacts-from-identites all-contacts public-key (keys community-members)))
+
+      :else (merge mentionable-users mentionable-contacts))))
 
 (defn get-mentionable-users
-  [{{:keys          [current-chat-id]
-     :contacts/keys [contacts] :as db} :db}]
-  (let [{:keys [chat-type community-id users] :as chat} (get-in db [:chats current-chat-id])
-        {:keys [name preferred-name public-key]}        (:multiaccount db)
-        chat-specific-suggestions
-        (cond
-          (= chat-type constants/private-group-chat-type)
-          (let [{:keys [public-key]} (:multiaccount db)
-                all-contacts (:contacts/contacts db)
-                group-contacts
-                (contact.db/get-all-contacts-in-group-chat
-                 (disj (:contacts chat) public-key)
-                 nil
-                 all-contacts
-                 nil)]
-            (reduce
-             (fn [acc {:keys [alias public-key identicon name]}]
-               (assoc acc public-key
-                      {:alias      alias
-                       :identicon  identicon
-                       :public-key public-key
-                       :name       name}))
-             {}
-             group-contacts))
-          (= chat-type constants/one-to-one-chat-type)
-          (assoc users
-                 current-chat-id
-                 (contact.db/public-key->contact contacts current-chat-id))
-
-          (= chat-type constants/community-chat-type)
-          (merge users (mentionable-commmunity-members
-                        contacts public-key
-                        (get-in db [:communities community-id :members])))
-
-          :else users)]
-    (reduce
-     (fn [acc [key {:keys [alias name identicon]}]]
-       (let [name (utils/safe-replace name ".stateofus.eth" "")]
-         (assoc acc key
-                {:alias      alias
-                 :name       (or name alias)
-                 :identicon  identicon
-                 :public-key key})))
-     (assoc chat-specific-suggestions
-            public-key
-            {:alias      name
-             :name       (or preferred-name name)
-             :public-key public-key})
-     contacts)))
+  [{:keys [db]}]
+  (let [current-chat-id      (:current-chat-id db)
+        chat                 (get-in db [:chats current-chat-id])
+        all-contacts         (:contacts/contacts db)
+        current-multiaccount (:multiaccount db)
+        community-members    (get-in db [:communities (:community-id chat) :members])]
+    (mentionable-users chat all-contacts current-multiaccount community-members)))
```

Comment:
why is an extra method needed?

Reply:
i mean why don't just have implementation here

Reply:
Hi @flexsurfer,
We are using mentionable users at two places
1. `src/status_im/subs.cljs L-1430`
2. `src/status_im/chat/models/mentions.cljs L-258`
Former provides required data using subscriptions and later using cofx. Due to this, earlier we used to have the same code at two places. Now with the help of this wrapper function, we are able to remove duplicate code and keep calculations of mentions in a single place. Thanks

Reply:
@flexsurfer, done 👍, Thanks